recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,197 @@
+import os
+import re
+
+from fnmatch import fnmatch
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class VTKConan(ConanFile):
+    name = "vtk"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://vtk.org/"
+    license = "BSD license"
+    description = "Visualization Toolkit by Kitware"
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    source_subfolder = "source_subfolder"
+    options = {"shared": [True, False], "with_qt": [True, False], "mpi": [True, False],
+                "fPIC": [True, False], "minimal": [True, False], "ioxml": [True, False],
+                "ioexport": [True, False], "mpi_minimal": [True, False]}
+    default_options = {"shared": False, "with_qt": False, "mpi": False, "fPIC": False,
+                "minimal": False, "ioxml": False, "ioexport": False, "mpi_minimal": False}
+    topics = ("conan", "VTK", "3D rendering", "2D plotting", "3D interaction", "3D manipulation", 
+                "graphics", "image processing", "scientific visualization", "geometry modeling")
+    short_paths = True
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Linux":
+            raise ConanInvalidConfiguration("VTK Conan package do not support Linux yet. Any help appreciated!")
+        if self.options.with_qt:
+            if self.options["qt"].shared == False:
+                raise ConanInvalidConfiguration("VTK option 'with_qt' requires 'qt:shared=True'")
+            if self.settings.os == "Linux":
+                if self.options["qt"].qtx11extras == False:
+                    raise ConanInvalidConfiguration("VTK option 'with_qt' requires 'qt:qtx11extras=True'")
```

Comment:
Options of external dependencies are not visible yet in `configure`. They are `None`.
(Try it for yourself by printing them here)
These checks should be moved to `build` until conan has a method to check the options (it is in the pipeline).

Reply:
Done